index.js

```diff
@@ -2,8 +2,67 @@
  * @format
  */
 
-import {AppRegistry} from 'react-native';
 import Login from './src/login';
+import Home from './src/home';
 import {name as appName} from './src/app.json';
+import React from 'react';
+import { View } from 'react-native';
+import { Navigation } from 'react-native-navigation';
+import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
 
-AppRegistry.registerComponent(appName, () => Login);
+// Initialize Apollo Client
+const client = new ApolloClient({
+  uri: 'https://tq-template-server-sample.herokuapp.com/graphql',
+  cache: new InMemoryCache()
+});
+  
+class RenderComponent extends React.Component {
+  render() {
+    return (
+      <View>
+        {this.props.render()}
+      </View>
+    );
+  }
+}
+
+function encapsulateComponent(Component) {
```

Comment:
booa! 
só uma coisa: geralmente, hoc's são nomeados com o prefixo `with`. então aqui seria algo assim
```suggestion
function withApollo(Component) {
```

Reply:
Ah, justo! Okay!